File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        Dictionary<int,int> map = new Dictionary<int, int>();
        
        for(int i = 0; i < nums.Length; i++){
            if(!map.ContainsKey(nums[i])){
                map[nums[i]] = 0;
            }

            map[nums[i]]++;
        }
        
        var sortedList = map.OrderByDescending(pair => pair.Value);
        var topK = sortedList.Take(k);

        return topK.Select(pair => pair.Key).ToArray();
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}

        for num in nums:
            freq[num] = 1 + freq.get(num, 0)

        sortedPair = sorted(freq.items(), key=lambda pair: pair[1], reverse = True)

        return [pair[0] for pair in sortedPair[:k]]